packages/contractkit/src/wrappers/MetaTransactionWallet.ts

```diff
@@ -0,0 +1,292 @@
+  | TransactionObjectWithValue<T>
+  | RawTransaction
+
+/**
+ * Class that wraps the MetaTransactionWallet
+ */
+export class MetaTransactionWalletWrapper extends BaseWrapper<MetaTransactionWallet> {
+  /**
+   * Execute a transaction originating from the MTW
+   * Reverts if the caller is not a signer
+   * @param tx a TransactionInput
+   */
+  public executeTransaction(tx: TransactionInput<any>): CeloTransactionObject<string> {
+    const rawTx = toRawTransaction(tx)
+    return toTransactionObject(
+      this.kit,
+      this.contract.methods.executeTransaction(rawTx.destination, rawTx.value, rawTx.data)
+    )
+  }
+
+  /**
+   * Execute a batch of transactions originating from the MTW
+   * Reverts if the caller is not a signer
+   * @param txs An array of TransactionInput
+   */
+  public executeTransactions(
+    txs: Array<TransactionInput<any>>
+  ): CeloTransactionObject<{ 0: string; 1: string[] }> {
+    const rawTxs: RawTransaction[] = txs.map(toRawTransaction)
+    const destinations = rawTxs.map((rtx) => rtx.destination)
+    const values = rawTxs.map((rtx) => rtx.value)
+    const callData = ensureLeading0x(rawTxs.map((rtx) => trimLeading0x(rtx.data)).join(''))
+    const callDataLengths = rawTxs.map((rtx) => trimLeading0x(rtx.data).length / 2)
+
+    return toTransactionObject(
+      this.kit,
+      this.contract.methods.executeTransactions(destinations, values, callData, callDataLengths)
+    )
+  }
+
+  /**
+   * Execute a signed meta transaction
+   * Reverts if meta-tx signer is not a signer for the wallet
+   * @param tx a TransactionInput
+   * @param signature a Signature
+   */
+  public executeMetaTransaction(
+    tx: TransactionInput<any>,
+    signature: Signature
+  ): CeloTransactionObject<string> {
+    const rawTx = toRawTransaction(tx)
+
+    return toTransactionObject(
+      this.kit,
+      this.contract.methods.executeMetaTransaction(
+        rawTx.destination,
+        rawTx.value,
+        rawTx.data,
+        signature.v,
+        signature.r,
+        signature.s
+      )
+    )
+  }
+
+  /**
+   * Signs a meta transaction as EIP712 typed data
+   * @param tx a TransactionWrapper
+   * @param nonce Optional -- will query contract state if not passed
+   * @returns signature a Signature
+   */
+  async signMetaTransaction(tx: TransactionInput<any>, nonce?: number): Promise<Signature> {
+    if (nonce === undefined) {
+      nonce = await this.nonce()
+    }
+    const typedData = buildMetaTxTypedData(
+      this.address,
+      toRawTransaction(tx),
+      nonce,
+      await this.chainId()
+    )
+    // TODO: This should be cached by the CeloProvider and executed through the wallets.
+    //       But I think the GethNativeBridgeWallet currently doesn't support this, it's
+    //       in the works.
+    const signer = await this.signer()
+    const signature = await new Promise<string>((resolve, reject) => {
+      ;(this.kit.web3.currentProvider as Pick<HttpProvider, 'send'>).send(
```

Comment:
Yeah agree I'll take another look to clean this up a bit.

Reply:
I'm not sure where the best place to move it would be, tho. I still haven't gotten a good sense of how utils are distributed in contractkit. I could see this being exposed as a ContractKit method so we could do:
```
kit.signTypedData(signer, typedData)
```

But I'm not sure if we're ok with adding that to ContractKit as a toplevel method.

Reply:
I made the change like this. I like the ergonomics on having that method exposed on `ContractKit` but I'll wait before merging this to get feedback on the choice.